pennylane/qaoa/cycle.py

```diff
@@ -417,28 +550,43 @@ def _inner_out_flow_constraint_hamiltonian(graph: nx.DiGraph, node) -> Hamiltoni
         ( \sum_{j,(i,j)\in E}\hat{Z}_{ij} )^{2}
 
     Args:
-        graph (nx.DiGraph): the directed graph specifying possible edges
+        graph (nx.DiGraph or rx.PyDiGraph): the directed graph specifying possible edges
         node: a fixed node
 
     Returns:
         qml.Hamiltonian: The inner part of the out-flow constraint Hamiltonian.
     """
+    if not isinstance(graph, (nx.DiGraph, rx.PyDiGraph)):
+        raise ValueError(
+            f"Input graph must be a nx.DiGraph or rx.PyDiGraph, got {type(graph).__name__}"
+        )
+
     coeffs = []
     ops = []
 
+    get_nvalues = (
+        lambda T: (graph.nodes().index(T[0]), graph.nodes().index(T[1]))
+        if isinstance(graph, rx.PyDiGraph)
+        else T
+    )
+
     edges_to_qubits = edges_to_wires(graph)
-    out_edges = graph.out_edges(node)
+    out_edges = sorted(graph.out_edges(node))
```

Comment:
I assume the sort here is necessary?

Reply:
nice catch, I missed this one!

Reply:
Yes indeed! One of the difficulties to get support for both RX and NX was that the results of some low-level routines aren't in the same order! Here, sorting the results of `out_edges` and `in_edges` is my trick to get equivalent results either using RX or NX.